### PR TITLE
Feature/smhe 1432 opvoeren alle gasmeter sleutels bij inladen shipment file g meter

### PR DESCRIPTION
**Pull request by natasja-n**
https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/53/

### CODE REVIEW

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java [hunk 1]
```diff
@@ -122,12 +120,55 @@ private List<SecurityKeyType> determineKeyTypesToStore(final SmartMeteringDevice
             ComponentType.PROTOCOL_DLMS,
             new IllegalArgumentException(msg));
       }
-      return Arrays.asList(G_METER_MASTER);
+      return gMeterKeyTypes;
```
Comment (kroesctrl) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java]: Should you not always all key types as done for e-meter

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SmartMeteringDevice.java [hunk 2]
```diff
@@ -36,6 +36,10 @@ public class SmartMeteringDevice implements Serializable {
   private byte[] globalEncryptionUnicastKey;
 
   private byte[] authenticationKey;
+  private byte[] globalEncryptionBroadcastKey;
```
Comment (kroesctrl) [inline osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SmartMeteringDevice.java]: Where do you use globalEncryptionBroadcastKey?

File: osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java [hunk 3]
```diff
@@ -36,6 +36,10 @@ public class SmartMeteringDeviceDto implements Serializable {
   private byte[] globalEncryptionUnicastKey;
 
   private byte[] authenticationKey;
+  private byte[] globalEncryptionBroadcastKey;
```
Comment (kroesctrl) [inline osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java]: Where do you use globalEncryptionBroadcastKey?

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/installation-ws-smartmetering.xsd [hunk 4]
```diff
@@ -332,6 +332,11 @@
         type="xsd:hexBinary" minOccurs="0" />
       <xsd:element name="Authentication_key" type="xsd:hexBinary"
         minOccurs="0" />
+      <xsd:element name="Global_encryption_broadcast_key" type="xsd:hexBinary" minOccurs="0" />
```
Comment (kroesctrl) [inline osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/installation-ws-smartmetering.xsd]: Where do you use Global_encryption_broadcast_key?

**Review by kroesctrl — CHANGES_REQUESTED** [review]
I would expect there is also a check in a cucumber file like: AddMBusDevice.feature to check if keys exist after request

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java (hunk 1 above)
Comment (natasja-n) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationService.java]: right, all key types are returned now.

File: osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SmartMeteringDevice.java (hunk 2 above)
Comment (natasja-n) [reply osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/valueobjects/smartmetering/SmartMeteringDevice.java]: The unused key is removed.

File: osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java (hunk 3 above)
Comment (natasja-n) [reply osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java]: The unused key is removed.

File: osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/installation-ws-smartmetering.xsd (hunk 4 above)
Comment (natasja-n) [reply osgp/shared/osgp-ws-smartmetering/src/main/resources/schemas/installation-ws-smartmetering.xsd]: The unused key is removed.

**Review by kroesctrl — APPROVED** [review]

**Pull request merged**